src/PimpMySql_Query.re

```diff
@@ -106,15 +120,21 @@ let updateOneById = (baseQuery, table, decoder, encoder, record, id, conn) => {
       |> PimpMySql_Params.positional
     );
   log("updateOneById", sql, params);
-  Sql.Promise.mutate(conn, ~sql, ~params?, ())
-  |> Js.Promise.then_(((success, _)) =>
-       if (success == 1) {
-         getOneById(baseQuery, table, decoder, id, conn)
-         |> Js.Promise.then_(res => Js.Promise.resolve(Result.pure(res)));
-       } else {
+  getOneById(baseQuery, table, decoder, id, conn)
+  |> Js.Promise.then_(res =>
+       switch (res) {
+       | Some(_) =>
+         Sql.Promise.mutate(conn, ~sql, ~params?, ())
+         |> Js.Promise.then_((_) =>
+              getOneById(baseQuery, table, decoder, id, conn)
+              |> Js.Promise.then_(res =>
```

Comment:
Can use the [Result.Promise.pure](https://github.com/scull7/bs-result/blob/master/src/Result.re#L105) method
to eliminate some of the march right code in the handler functions..
```reason
getOneById(baseQuery, table, decoder, id, conn)
|> Js.Promise.then_(
  fun
  | Some(_) =>
    Sql.Promise.mutate(conn, ~sql, ~params?, ())
    |> Js.Promise.then_((_) => getOneById(baseQuery, table, decoder, id, conn))
    |> Js.Promise.then_(Result.Promise.pure)
  | None => PimpMySql.Error.NotFound("Error: updateOneById failed") |> Result.Promise.error
)
```

Reply:
The `option('a) => Result.t(PimpMySql_Error.NotFound(string), 'a)` pattern seems very common, should we make a helper function to eliminate some of the boilerplate code?

```reason
maybeNotFound = (msg, maybe) =>
  switch (maybe) {
  | Some(x) => Result.pure(x)
  | None => PimpMySql.Error.NotFound(msg) |> Result.error
  };

thenMaybeNotFound = (msg, promise) =>
  promise
  |> Js.Promise.then_((x) => maybeNotFound(msg, x) |> Js.Promise.resolve);

getOneById(baseQuery, table, decoder, id, conn)
|> thenMaybeNotFound("Error: updateOneById failed")
|> Result.Promise.andThen((_) =>
    Sql.Promise.mutate(conn, ~sql, ~params?, ())
    |> Js.Promise.then_((_) => getOneById(baseQuery, table, decoder, id, conn))
    |> Js.Promise.then_(Result.Promise.pure)
)
```

Reply:
Will make these changes in the latest PR.